packages/flutter/lib/src/material/dialog.dart

```diff
@@ -85,6 +97,7 @@ class Dialog extends StatelessWidget {
               color: _getColor(context),
               type: MaterialType.card,
               child: child,
+              shape: shape ?? _defaultDialogShape,
```

Comment:
null should mean "no shape", i.e. just a rectangle.

Reply:
That does sound reasonable, but I think we need the `null` check to be our version of "did they pass in a custom shape? If not use, the theme's shape, finally use the default shape". For Material Dialogs, I think the default should be a border radius of 4.0 (matching the material guideline).

Reply:
OK, that's reasonable and consistent with the way other classes handle unspecified visual parameters.